### DISCUSSION TITLE
Internationalization

### DISCUSSION THREAD

**Discussion by Valdes-Tresanco-MS** [reflex-dev/reflex#2897] [Q&A]
Is there a way to do internationalization?
I certainly can't find a way to wrap `react-i18next` or `react-intl`
Any example can help.
Thanks in advance.

**Comment by picklelo** [comment]
Would something like this work?

```python
class Trans(rx.Component):
    """I18n component."""

    # The React library to wrap.
    library = 'react-i18next'

    # The React component tag.
    tag = "Trans"
    
trans = Trans.create
```

And then use the function like

```python
def index():
  return trans("hello")
```

I'm not an expert on this, but from my glance at the docs, the Trans React component seems to do this. https://react.i18next.com/latest/trans-component

**Reply by masenf** [reply]
I think it would take a little more than just that, but this looks like a promising approach to deliver internationalization as a separate installable component.
